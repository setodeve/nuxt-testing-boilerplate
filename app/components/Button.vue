<template>
  <p data-testid="count">{{ cnt }}</p>
  <button @click="emit('click')">{{ props.msg }}</button>
</template>

<script setup lang="ts">
import { ref } from 'vue'
import type { Ref } from 'vue'
const props = defineProps({
  msg: String,
  cnt: Number as Ref<number>
})
const emit = defineEmits(['click'])
</script>
