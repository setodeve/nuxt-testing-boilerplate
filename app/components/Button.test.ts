import { describe, it, expect } from 'vitest'
import { mount } from '@vue/test-utils'
import Button from './Button.vue'
import { ref } from 'vue'

describe('Button Component', () => {
  const cnt = ref(0)
  
  it('renders count', () => {
    const wrapper = mount(Button, {
      props: { msg: 'Hello', cnt: cnt },
    })
    const count = wrapper.find('[data-testid="count"]')
    expect(count.text()).toBe('0')
  })

  it('emits click event', async () => {
    const cntUp = (count: number) => count + 1
    const wrapper = mount(Button, {
      props: { msg: 'Hello', cnt: cnt },
      methods: { cntUp: cntUp },
    })
    await wrapper.find('button').trigger('click')
    expect(wrapper.emitted()).toHaveProperty('click')
  })

  it('matches snapshot', () => {
    const wrapper = mount(Button, {
      props: { msg: 'Hello', cnt: cnt },
    })
    expect(wrapper.html()).toMatchSnapshot()
  })
})