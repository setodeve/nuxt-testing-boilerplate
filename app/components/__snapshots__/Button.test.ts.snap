// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Button Component > matches snapshot 1`] = `
"<p data-testid="count">0</p>
<button>Hello</button>"
`;
